File: ChMS.Web/Controllers/IncomeController.cs
using Microsoft.AspNetCore.Mvc;
using Chms.Application.Common.Interface;
using Chms.Domain.Entities;
using Microsoft.AspNetCore.Authorization;
using ChMS.Web.ViewModels;
using Income = Chms.Domain.Entities.Income;
using Chms.Domain.ViewModels.IncomeExpense;
using Microsoft.AspNetCore.Identity;
using Microsoft.VisualBasic;

// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860

namespace ChMS.Web.Controllers
{
    [Route("[controller]")]
    [ApiController]
    [Authorize]
    public class IncomeController : ControllerBase
    {
        public readonly IIncomeService _service;
        public readonly IFileUploadService _uploadService;
        private readonly ICurrentUserService _currentUserService;
        public IncomeController(IIncomeService service, IFileUploadService uploadService, ICurrentUserService currentUserService)
        {
            _service = service;
            _uploadService = uploadService;
            _currentUserService = currentUserService;
        }


        [HttpPost]
        public async Task<ActionResult<int>> Create([FromBody] AddIncome request)
        {
            try
            {
                if (request.Income.Count == 0 || String.IsNullOrEmpty(request.Date.ToString()))
                {
                    return Ok("Nothing found");
                }

                foreach (var income in request.Income)
                {
                    Income entity = new()
                    {
                        Category = income.Category,
                        Amount = income.Amount,
                        IncomeDate = request.Date,
                        MemberId = income.MemberId,
                        Description = income.Description,
                        CreatedBy = _currentUserService.UserId
                    };
                    await _service.Create(entity);
                }

                return Ok();

            }
            catch (Exception ex)
            {
                return BadRequest(ex.Message);
            }
           
        }


        [HttpPut]
        public async Task<ActionResult> Update([FromBody] EditIncome request)
        {
            if (request.Income.Count == 0 || String.IsNullOrEmpty(request.Date.ToString()))
            {
                return BadRequest("Invalid Input Data");
            }
            try
            {
                foreach (var income in request.Income)
                {
                    Income entity = new()
                    {
                        Id = income.Id,
                        Category = income.Category,
                        Amount = income.Amount,
                        IncomeDate = request.Date,
                        MemberId = income.MemberId,
                        Description = income.Description,
                        UpdatedDate = DateTime.Now,
                        UpdatedBy = _currentUserService.UserId
                    };
                    await _service.Update(entity);
                }

                return Ok();

            }
            catch (Exception ex)
            {
                return BadRequest(ex.Message);
            }
        }

        [HttpDelete("{incomeDate}")]
        public ActionResult Delete(string incomeDate)
        {
            if (String.IsNullOrEmpty(incomeDate.ToString()))
            {
                return BadRequest("Income Date is not selected");
            }
            try
            {
                _service.Delete(incomeDate);
                return Ok();
            }
            catch (Exception ex)
            {
                return BadRequest(ex.Message);
            }

        }

        [HttpGet("{incomeDate}")]
        public ActionResult Get(string incomeDate)
        {
            if(String.IsNullOrEmpty(incomeDate.ToString()))
            {
                return BadRequest("Income Date is not selected");
            }
            try
            {
                return Ok(_service.Get(incomeDate));
            }
            catch (Exception ex)
            {
                return BadRequest(ex.Message);
            }
        }

        [HttpGet("list")]
        public ActionResult List([FromQuery] IncomeFilterQueryVm filterQuery)
        {
            var date = DateTime.Now;
            var firstDayOfMonth = new DateTime(date.Year, date.Month, 1);
            var lastDayOfMonth = firstDayOfMonth.AddMonths(1).AddDays(-1);
            filterQuery.StartDate ??= firstDayOfMonth.ToString("yyyy-MM-dd");
            filterQuery.EndDate ??= lastDayOfMonth.ToString("yyyy-MM-dd");

            FilterVm filterVm = new(){
                StartDate = filterQuery.StartDate,
                EndDate = filterQuery.EndDate,
            };
            return Ok(_service.List(filterVm));
        }


        [HttpGet("total-income")]
        public ActionResult TotalCount([FromQuery] IncomeFilterQueryVm filterQuery)
        {
            var date = DateTime.Now;
            var firstDayOfMonth = new DateTime(date.Year, date.Month, 1);
            var lastDayOfMonth = firstDayOfMonth.AddMonths(1).AddDays(-1);
            filterQuery.StartDate ??= firstDayOfMonth.ToString("yyyy-MM-dd");
            filterQuery.EndDate ??= lastDayOfMonth.ToString("yyyy-MM-dd");

            FilterVm filterVm = new(){
                StartDate = filterQuery.StartDate,
                EndDate = filterQuery.EndDate,
            };
            return Ok(_service.TotalDataCount(filterVm));
        }
    }
}

File: API/app/Http/Controllers/IncomeController.php
<?php

namespace App\Http\Controllers;

use App\Models\Income;
use App\Http\Requests\Request;
use Carbon\Carbon;
use Illuminate\Support\Facades\Auth;

class IncomeController extends Controller
{
    // POST: Create Incomes
    public function create(Request $request)
    {
        try {
            if (empty($request->income) || empty($request->date)) {
                return response()->json("Nothing found");
            }

            foreach ($request->income as $income) {
                Income::create([
                    'category'     => $income['category'] ?? null,
                    'amount'       => $income['amount'] ?? 0,
                    'income_date'  => $request->date,
                    'member_id'    => $income['member_id'] ?? null,
                    'description'  => $income['description'] ?? null,
                    'created_by'   => Auth::id(),
                ]);
            }

            return response()->json(['message' => 'Created successfully']);
        } catch (\Exception $ex) {
            return response()->json(['error' => $ex->getMessage()], 400);
        }
    }

    // PUT: Update Incomes
    public function update(Request $request)
    {
        if (empty($request->income) || empty($request->date)) {
            return response()->json("Invalid Input Data", 400);
        }

        try {
            foreach ($request->income as $income) {
                $incomeModel = Income::find($income['id']);
                if ($incomeModel) {
                    $incomeModel->update([
                        'category'     => $income['category'] ?? null,
                        'amount'       => $income['amount'] ?? 0,
                        'income_date'  => $request->date,
                        'member_id'    => $income['member_id'] ?? null,
                        'description'  => $income['description'] ?? null,
                        'updated_by'   => Auth::id(),
                        'updated_at'   => now(),
                    ]);
                }
            }

            return response()->json(['message' => 'Updated successfully']);
        } catch (\Exception $ex) {
            return response()->json(['error' => $ex->getMessage()], 400);
        }
    }

    // DELETE: Delete by date
    public function delete($incomeDate)
    {
        if (empty($incomeDate)) {
            return response()->json("Income Date is not selected", 400);
        }

        try {
            Income::where('income_date', $incomeDate)->delete();
            return response()->json(['message' => 'Deleted successfully']);
        } catch (\Exception $ex) {
            return response()->json(['error' => $ex->getMessage()], 400);
        }
    }

    // GET: Get by date
    public function get($incomeDate)
    {
        if (empty($incomeDate)) {
            return response()->json("Income Date is not selected", 400);
        }

        try {
            $data = Income::where('income_date', $incomeDate)->get();
            return response()->json($data);
        } catch (\Exception $ex) {
            return response()->json(['error' => $ex->getMessage()], 400);
        }
    }

    // GET: List with filter
    public function list(Request $request)
    {
        $date = Carbon::now();
        $firstDayOfMonth = $date->copy()->startOfMonth()->format('Y-m-d');
        $lastDayOfMonth  = $date->copy()->endOfMonth()->format('Y-m-d');

        $startDate = $request->query('start_date', $firstDayOfMonth);
        $endDate   = $request->query('end_date', $lastDayOfMonth);

        $data = Income::whereBetween('income_date', [$startDate, $endDate])->get();

        return response()->json($data);
    }

    // GET: Total Income Count
    public function total(Request $request)
    {
        $date = Carbon::now();
        $firstDayOfMonth = $date->copy()->startOfMonth()->format('Y-m-d');
        $lastDayOfMonth  = $date->copy()->endOfMonth()->format('Y-m-d');

        $startDate = $request->query('start_date', $firstDayOfMonth);
        $endDate   = $request->query('end_date', $lastDayOfMonth);

        $total = Income::whereBetween('income_date', [$startDate, $endDate])->sum('amount');

        return response()->json(['total_income' => $total]);
    }
}
